src/reducer.js

```diff
@@ -280,6 +288,50 @@ function reducer(
         fetching_obligatory_eo_fields: false,
         errorL1s: formatServerError(action.payload),
       };
+    case "USERNAME_VALIDATION_FIELDS_REQ":
+      return {
+        ...state,
+        validationFields: {
+          username: {
+            isValidating: true,
+            isValid: false,
+            validationError: null,
+          },
+        },
+      };
+    case "USERNAME_VALIDATION_FIELDS_RESP":
+      return {
+        ...state,
+        validationFields: {
```

Comment:
It might be a good idea to spread here the state.validationFields. It would be useful when we want to validate more fields in the same form. Take a look at the LOCATION_VALIDATION_FIELDS reducers in Location module.